### PR TITLE
Add documentation skeleton

### PR DESCRIPTION
**Pull request by stephenworsley**
Addresses #264

### CODE REVIEW

File: docs/src/installing.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+==========
+
+Iris is available using conda for the following platforms:
+
+* Linux 64-bit,
+* Mac OSX 64-bit, and
+* Windows 64-bit.
+
+.. note:: Iris-esmf-regrid is currently supported and tested against
+          |python_support| running on Linux.  We do not currently
```
Comment (trexfeathers) [inline docs/src/installing.rst:11]:
```
reading sources... [100%] whatsnew/index                                                                                      
/net/home/h01/myeo/repos/iris-esmf-regrid/docs/src/installing.rst:10: ERROR: Undefined substitution referenced: "python_support".
```

**Comment by trexfeathers** [comment]
Might be worth committing a `docs/src/_static` directory, since it's referenced in `conf.py` and that has been causing a warning during docs building for the last year. Either that or you could remove the entry in `conf.py`.

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/9a0b5571552f345315d0f9f795b5494de0e6208f/docs/src/conf.py#L89-L92

File: docs/src/index.rst [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
-A collection of structured and unstructured ESMF_ regridding schemes for Iris_.
+Iris-esmf-regrid is a package which aims to bridge the gap between Iris_ data
+handling and ESMF_ regridding. This is done largely by way of providing
+Iris_-like regridding schemes.
 
-This rendered documentation is so far limited to iris-esmf-regrid's API. Some
-basic examples and change logs can be found on GitHub_.
+This documentation is a work in progress and some pages will be incomplete.
 
 :doc:`Click to view the API<_api_generated/modules>`
```
Comment (trexfeathers) [inline docs/src/index.rst:10]: Is this still appropriate now there are multiple documentation sections? There is also a link in the top bar.

image

**Review by trexfeathers — COMMENTED** [review]
The title of the documentation includes `0.6.0`, which is misleading when viewing bleeding-edge documentation. For the last two releases we've forgotten to set `__version__` to be a dev version immediately after the release, like we did in #157.

File: docs/src/index.rst (hunk 2 above)
Comment (stephenworsley) [reply docs/src/index.rst:10]: yeah, I'm still thinking of how to arrange links within the index page, I'm thinking I might want a little bit of prose with particalar links in them.

File: docs/src/index.rst [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 
 .. toctree::
    :hidden:
 
+   installing
+   userguide/index
+   developers_guide/index
    _api_generated/modules
+   whatsnew/index
```
Comment (trexfeathers) [inline docs/src/index.rst:34]: This new documentation is our first opportunity to try the structure laid out in the [Divio documentation system](https://documentation.divio.com/).

I like the heading titles in the [Matplotlib docs](https://matplotlib.org/stable/index.html), which are one example of implementing the Divio system:

image

File: docs/src/index.rst [hunk 4]
```diff
@@ -1,18 +1,22 @@
 iris-esmf-regrid
 ================
 
-A collection of structured and unstructured ESMF_ regridding schemes for Iris_.
+Iris-esmf-regrid is a package which aims to bridge the gap between Iris_ data
+handling and ESMF_ regridding. This is done largely by way of providing
+Iris_-like regridding schemes.
 
-This rendered documentation is so far limited to iris-esmf-regrid's API. Some
-basic examples and change logs can be found on GitHub_.
```
Comment (trexfeathers) [inline docs/src/index.rst:7]: If there is no clear 'division of labour' between the full documentation and the `README`, I suggest migrating everything over to the full documentation. We don't want a situation where both need to be kept up-to-date in tandem.

We can leave anything in the `README` that doesn't have a defined place elsewhere.

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Here are the rest of my comments 🙂

File: docs/src/index.rst
[the remark this answers is not in the document]
Comment (trexfeathers) [reply docs/src/index.rst:1]: Looking good now!

File: docs/src/userguide/examples.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+Note that this scheme is flexible and it is also possible to regrid from
+a structured cube to an unstructured cube as follows::
+
+    result = target_grid_cube.regrid(source_mesh_cube, ESMFAreaWeighted())
+
+Saving and Loading a Regridder
+------------------------------
+A regridder can be set up for reuse, this saves time performing the
+computationally expensive initialisation process::
+    from esmf_regrid.experimental.unstructured_scheme import MeshToGridESMFRegridder
```
Comment (trexfeathers) [inline docs/src/userguide/examples.rst:33]: I'm getting a build warning here - think it needs an extra line break?

```
.../iris-esmf-regrid/docs/src/userguide/examples.rst:33: ERROR: Unexpected indentation.
```

Comment (trexfeathers) [reply docs/src/userguide/examples.rst:34]: Building fine now 👍

File: docs/src/index.rst (hunk 3 above)
Comment (stephenworsley) [reply docs/src/index.rst:41]: As far as the overall structure goes, I believe that the Divio system would make most sense as something to apply structure *along side* the existing structure of the heading bar. For example, with the API existing as its own clearly labeled thing in the heading bar but also listed as reference documentation with respect to the Divio system. I've rewritten the index page to make the more clear.

**Review by trexfeathers — APPROVED** [review]

**Pull request merged**